src/amo/pages/UserProfileEdit/index.js

```diff
@@ -198,8 +200,25 @@ export class UserProfileEditBase extends React.Component<Props, State> {
       });
     }
 
-    if (wasUpdating && !isUpdating && !errorHandler.hasError()) {
+    const redirectToUserProfile = () => {
       history.push(`/${lang}/${clientApp}/user/${newUserId}/`);
+    };
+
+    if (wasUpdating && !isUpdating && !errorHandler.hasError()) {
+      let toPath = location.query.to;
+      if (toPath) {
+        if (!toPath.startsWith('/')) {
+          toPath = `/${toPath}`;
+        }
+        try {
+          history.push(toPath);
+        } catch (error) {
+          log.warn(`Error redirecting to location: ${toPath}`);
+          redirectToUserProfile();
+        }
+      } else {
+        redirectToUserProfile();
```

Comment:
I think it would be safer to return from this function after the redirect. Some `window.scroll()` code might run which is probably harmless but there could be future surprises.